mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -4265,3 +4264,160 @@ fn test_utxo_validate_valid_and_invalid_pubkey() {
     assert!(coin.validate_other_pubkey(&[1u8; 20]).is_err());
     assert!(coin.validate_other_pubkey(&[1u8; 8]).is_err());
 }
+
+#[test]
+fn test_block_header_utxo_loop() {
+    static mut CURRENT_BLOCK_COUNT: u64 = 2048;
+
+    ElectrumClient::get_block_count
+        .mock_safe(|_| MockResult::Return(Box::new(futures01::future::ok(unsafe { CURRENT_BLOCK_COUNT }))));
+
+    let block_header_storage = BlockHeaderStorageForTests::new(TEST_COIN_NAME.to_string());
+
+    let block_header_storage_copy = block_header_storage.clone();
+    BlockHeaderStorage::new_from_ctx.mock_safe(move |_, _| {
+        MockResult::Return(Ok(BlockHeaderStorage {
+            inner: Box::new(block_header_storage_copy.clone()),
+        }))
+    });
+
+    let block_header_storage_copy = block_header_storage.clone();
+    BlockHeaderStorage::add_block_headers_to_storage.mock_safe(move |_, headers| {
```

Comment:
This doesn't make sense because:
1) `BlockHeaderStorageForTests` already does exactly the same as you described here
2) `impl BlockHeaderStorageOps for BlockHeaderStorageForTests` is not marked as `#[mockable]`. This mocked function is never fired. I even tried to put a panic here to make sure :)